### PR TITLE
get_job_def Response xml/yaml and get_job_meta Response xml methods added to help on Rundeck researches etc

### PR DESCRIPTION
**Pull request by msilva14**
get_job_def Response xml/yaml and get_job_meta Response xml methods added to help on Rundeck researches etc

### CODE REVIEW

File: pyrundeck/rundeck.py [hunk 1]
```diff
@@ -77,15 +77,18 @@ def __request(self, method, url, params=None, upload_file=None):
...
+            except ValueError as e:
+                logger.error(e.message)
+                return r.content
+        else:
+           return r.text
 
-    def __get(self, url, params=None):
-        return self.__request("GET", url, params)
+    def __get(self, url, params=None, format='json'):
+        return self.__request("GET", url, params, format=format)
```
Comment (pschmitt) [inline pyrundeck/rundeck.py:91]: Here I'd like to see a check of the possible `format` values. The API won't know how to deal with something other than XML or json.

Raising an exception then would be ideal.

**Review by pschmitt — CHANGES_REQUESTED** [review]
Thanks for the PR. There's a few things I'd like to see changed, but other than that: LGTM!

**Comment by pschmitt** [comment]
Oh and please run `black` on your code!

File: pyrundeck/rundeck.py [hunk 2]
```diff
@@ -77,15 +77,18 @@ def __request(self, method, url, params=None, upload_file=None):
...
         r = requests.request(method, url, **options)
         logger.debug(r.content)
-        r.raise_for_status()
-        try:
-            return r.json()
-        except ValueError as e:
-            logger.error(e.message)
-            return r.content
+        if format == 'json':
+            r.raise_for_status()
```
[the remark this answers is not in the document]
Comment (msilva14) [reply pyrundeck/rundeck.py:81]: Ok, it's done.... when adding if I let it inside...

File: pyrundeck/rundeck.py (hunk 1 above)
Comment (msilva14) [reply pyrundeck/rundeck.py:94]: Thanks for advising it.. format tested and exception raised..

**Review by pschmitt — APPROVED** [review]

**Comment by pschmitt** [comment]
Thanks :)

**Comment by pschmitt** [comment]
This is now part of 0.9.13: https://pypi.org/project/pyrundeck/0.9.13/

**Pull request merged**